fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/transformation/EliminateGroupByConstantRule.java

```diff
@@ -0,0 +1,102 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.sql.optimizer.rule.transformation;
+
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import com.starrocks.sql.optimizer.OptExpression;
+import com.starrocks.sql.optimizer.OptimizerContext;
+import com.starrocks.sql.optimizer.operator.OperatorType;
+import com.starrocks.sql.optimizer.operator.logical.LogicalAggregationOperator;
+import com.starrocks.sql.optimizer.operator.logical.LogicalProjectOperator;
+import com.starrocks.sql.optimizer.operator.pattern.Pattern;
+import com.starrocks.sql.optimizer.operator.scalar.ColumnRefOperator;
+import com.starrocks.sql.optimizer.operator.scalar.ScalarOperator;
+import com.starrocks.sql.optimizer.rewrite.ReplaceColumnRefRewriter;
+import com.starrocks.sql.optimizer.rule.RuleType;
+
+import java.util.List;
+import java.util.Map;
+
+public class EliminateGroupByConstantRule extends TransformationRule {
+
+    public static final EliminateGroupByConstantRule INSTANCE = new EliminateGroupByConstantRule();
+
+    private EliminateGroupByConstantRule() {
+        super(RuleType.TF_ELIMINATE_GROUP_BY_CONSTANT,
+                Pattern.create(OperatorType.LOGICAL_AGGR, OperatorType.LOGICAL_PROJECT));
+    }
+
+    @Override
+    public boolean check(OptExpression input, OptimizerContext context) {
+        LogicalAggregationOperator aggOp = input.getOp().cast();
+        LogicalProjectOperator projectOp = input.inputAt(0).getOp().cast();
+        Map<ColumnRefOperator, ScalarOperator> columnRefMap = projectOp.getColumnRefMap();
+        if (aggOp.getGroupingKeys().isEmpty()) {
+            return false;
+        }
+
+        for (ColumnRefOperator groupByCol : aggOp.getGroupingKeys()) {
+            if (!columnRefMap.get(groupByCol).isConstant()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    @Override
+    public List<OptExpression> transform(OptExpression input, OptimizerContext context) {
+        LogicalAggregationOperator aggOp = input.getOp().cast();
+        LogicalProjectOperator projectOp = input.inputAt(0).getOp().cast();
+        Map<ColumnRefOperator, ScalarOperator> bottomMap = Maps.newHashMap();
+        Map<ColumnRefOperator, ScalarOperator> topMap = Maps.newHashMap();
+        for (Map.Entry<ColumnRefOperator, ScalarOperator> entry : projectOp.getColumnRefMap().entrySet()) {
+            if (aggOp.getGroupingKeys().contains(entry.getKey())) {
+                topMap.put(entry.getKey(), entry.getValue());
+            } else {
+                bottomMap.put(entry.getKey(), entry.getValue());
+            }
+        }
+
+        aggOp.getAggregations().keySet().forEach(e -> topMap.put(e, e));
```

Comment:
should check groupby is empty, must save one at least. BTW, I think we need pull all constant value rule? like deriver constant value

Reply:
This rule only process all group by keys are constant.  So it's correct to take this scene as a non group by agg.

Reply:
Yes, I agree with you. We need a such rule. Let me finish it later.

Reply:
> This rule only process all group by keys are constant. So it's correct to take this scene as a non group by agg.

yeah, group by a column is differenet with no group by when the rows empty, so need save one at least

Reply:
btw, sr will remove constants value in transformer phase now, you can let it remove all constant in group by, not only `all group by keys are constant`.

Reply:
a other question, why sr will generate `agg(update serialize)-project-agg(update serialize)`? it's a invalid plan...and we should fix the count's nullable info?